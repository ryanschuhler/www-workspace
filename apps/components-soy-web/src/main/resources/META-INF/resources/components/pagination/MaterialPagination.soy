{namespace MaterialPagination}

/**
 * MaterialPagination
 */
{template .render}
  {@param end: string|null}
  {@param portletNamespace: string}
  {@param start: string|null}
  {@param total: string|null}
  {@param? cssClass: string}
  {@param? nextURL: string|null}
  {@param? previousURL: string|null}
  {@param? tooltipCssClass: string}

  {let $namespace: $portletNamespace + randomInt(1000) /}

  <div class="pario-pagination-container {$cssClass ?: ''}">
		<span class="pario-pagination-previous pario-pagination-url">
			{if $previousURL}
      <a href="{$previousURL}" id="{$namespace}_previous">
      {/if}

      <i class="material-icons material-icons-middle">chevron_left</i>

      {if $previousURL}
      </a>

        <div class="mdl-tooltip {$tooltipCssClass ?: ''}" data-mdl-for="{$namespace}_previous">
          {msg desc=""}previous-page{/msg}
        </div>
      {/if}
		</span>

    {$start}{sp}-{sp}{$end}

    {sp}

    {msg desc=""}of{/msg}

    {sp}

    {$total}

    <span class="pario-pagination-next pario-pagination-url">
			{if $nextURL}
      <a href="{$nextURL}" id="{$namespace}_next">
      {/if}

      <i class="material-icons material-icons-middle">chevron_right</i>

      {if $nextURL}
      </a>

        <div class="mdl-tooltip {$tooltipCssClass ?: ''}" data-mdl-for="{$namespace}_next">
          {msg desc=""}next-page{/msg}
        </div>
      {/if}
		</span>
  </div>
{/template}
