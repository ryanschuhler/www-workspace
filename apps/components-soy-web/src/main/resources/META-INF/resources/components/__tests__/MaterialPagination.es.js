'use strict';

import MaterialPagination from '../pagination/MaterialPagination.es';

let component;

describe('MaterialPagination', () => {
  afterEach(() => {
    if (component) {
      component.dispose();
    }
  });

  it('should render', () => {
    component = new MaterialPagination({
      end: '10',
      portletNamespace: 'foo',
      start: '0',
      total: '20'
    });

    expect(component).toMatchSnapshot();
  });
});