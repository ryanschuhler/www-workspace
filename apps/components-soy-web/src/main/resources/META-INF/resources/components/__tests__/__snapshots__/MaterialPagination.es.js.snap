// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MaterialPagination should render 1`] = `
<div class="pario-pagination-container ">
  <span class="pario-pagination-previous pario-pagination-url">
    <i class="material-icons material-icons-middle">chevron_left</i>
  </span>0 - 10 of 20
  <span class="pario-pagination-next pario-pagination-url">
    <i class="material-icons material-icons-middle">chevron_right</i>
  </span>
</div>
`;
